rust/main/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -677,14 +677,19 @@ async fn confirm_lander_task(
             .map(|(op, result)| async {
                 let message_id = op.id();
                 match result {
-                    Ok(Some(payload_id)) => Some((op, entrypoint.payload_status(payload_id).await)),
-                    Ok(None) | Err(_) => {
+                    Ok(Some(ids)) if !ids.is_empty() => Some((
+                        op,
+                        entrypoint
+                            .payload_status(ids.into_iter().last().unwrap())
+                            .await,
+                    )),
```

Comment:
IMO all payloads linked to a message must be confirmed, as a requirement for the message to also count as confirmed. So here we should check the status for all payloads rather than just the last one

Reply:
It is part of the next task: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5894

Reply:
Fixed